src/Http/Routing/src/Matching/DfaMatcher.cs

```diff
@@ -68,6 +69,26 @@ public sealed override Task MatchAsync(HttpContext httpContext, EndpointSelector
                 Logger.CandidatesFound(_logger, path, candidates);
             }
 
+            var policyCount = policies.Length;
+
+            // This is a fast path for single candidate, 0 policies and default selector
+            if (policyCount == 0 && candidateCount == 1 && _selector is DefaultEndpointSelector)
+            {
+                ref var candidate = ref candidates[0];
+                var flags = candidate.Flags;
```

Comment:
The flags are only used once. Is there any benefit from a local variable?

Reply:
Prolly not.